### PR TITLE
Restricts MatchedRule list to rules with log directive

### PR DESCRIPTION
**Pull request by M4tteoP**
closes https://github.com/corazawaf/coraza/issues/839

### CODE REVIEW

File: internal/corazawaf/transaction.go [hunk 1]
```diff
@@ -505,9 +505,12 @@ func (tx *Transaction) MatchRule(r *Rule, mds []types.MatchData) {
 		}
 	}
 
-	tx.matchedRules = append(tx.matchedRules, mr)
-	if tx.WAF.ErrorLogCb != nil && r.Log {
-		tx.WAF.ErrorLogCb(mr)
+	if r.Log {
+		// Only rules with log data will added to the matchedRules list
```
Comment (M4tteoP) [inline internal/corazawaf/transaction.go:509]: Main change

File: internal/corazawaf/transaction.go [hunk 2]
```diff
@@ -505,9 +505,12 @@ func (tx *Transaction) MatchRule(r *Rule, mds []types.MatchData) {
...
+		tx.matchedRules = append(tx.matchedRules, mr)
```
Comment (M4tteoP) [inline internal/corazawaf/transaction.go:510]: Main change

**Review by jptosso — APPROVED** [review]
I expected more difficulties for this change 😅 lgtm

**Comment by M4tteoP** [comment]
@jptosso That's why I opened it as a draft, I want to double-check that it is "just this" 😂

**Review by jcchavezs — APPROVED** [review]

**Comment by jcchavezs** [comment]
Can we turn this into ready for review? also would like to have a second eye from @airween @dune73

**Comment by dune73** [comment]
I can not contribute with a code review, but if you can explain me what's up, I can at least comment from a conceptional perspective. Please start by explaining "matchdata list".

**Comment by M4tteoP** [comment]
Hey @dune73, thanks for chiming in. "matchdata list" in the title was actually not the right term (updated now), it rather is currently the `MatchedRule` list, a list of all the rules that matched during a transaction.
A user running CRS v4.0 rc1 [reported](https://github.com/corazawaf/coraza/discussions/833#discussioncomment-6374355) that this list (which can be accessed by the connector) is filled by many rules related to setting variables and configuring CRS elements (E.g. any rule related to checking the `TX:EXECUTING_PARANOIA_LEVEL`).
The [idea](https://github.com/corazawaf/coraza/discussions/833#discussioncomment-6374388) which this PR implements, is to include in this list only the rules marked with a `log` action. Just like @jptosso told a few [comments above ](https://github.com/corazawaf/coraza/pull/840#issuecomment-1624078403): this list would become a list of **logged rules** rather than the **full list of matched rules**

**Comment by dune73** [comment]
Got you. I presume this list is coraza-specific and that's why I do not know it?

Based on your explanation, I presume that a blocking rule with `nolog` would not make it into the list. Could that lead to problems?

File: internal/corazawaf/transaction.go (hunk 2 above)
Comment (joshi-mohit) [reply internal/corazawaf/transaction.go:512]: Should we not return from the top ... if r.Log is not there why to do even some processing (a debug log will help for those cases where a rule has an action but log -- possibly will be atypo only :))   -- cannot see anything obvious breaking here if we exit right from start

**Comment by jcchavezs** [comment]
@M4tteoP could you please backchannel @dune73 on this?

**Comment by M4tteoP** [comment]
Giving to it a second thought, I don't think that adding an exception like this altering the concept of the `MatchedRule` list is great. Just like Christian's corner case, there might be others that we might don't take into account leading to misunderstandings or wrong behaviors.

Following an offline conversation with @jptosso, this PR is going to be superseded by https://github.com/corazawaf/coraza/pull/848. We aim to keep `MatchedRule` as it is but rather provide ways to filter the matched rules for advanced use cases. Looking forward to feedback on it!

**Pull request closed without merging**